lib/Optimizer/Transforms/GenDeviceCodeLoader.cpp

```diff
@@ -44,10 +44,11 @@ class GenerateDeviceCodeLoader
       // Add declaration of deviceCodeHolderAdd
       builder.create<LLVM::LLVMFuncOp>(
           loc, "deviceCodeHolderAdd",
-          LLVM::LLVMFunctionType::get(
-              cudaq::opt::factory::getVoidType(ctx),
-              {cudaq::opt::factory::getPointerType(ctx),
-               cudaq::opt::factory::getPointerType(ctx)}));
+          LLVM::LLVMFunctionType::get(cudaq::opt::factory::getVoidType(ctx),
+                                      {cudaq::opt::factory::getPointerType(ctx),
+                                       cudaq::opt::factory::getPointerType(ctx),
+                                       // bool --> char type (int 8)?
+                                       cudaq::opt::factory::getCharType(ctx)}));
```

Comment:
Can just use `builder.getI1Type()` here.

Reply:
You have to change the code where the call is made as well, and add a `ConstantIntOp` of `i1 0` as the final argument. Otherwise, the code will just pass garbage from the stack at runtime.